sqlmesh/core/engine_adapter/base.py

```diff
@@ -955,6 +960,23 @@ def create_view(
         if not self.SUPPORTS_VIEW_SCHEMA and isinstance(schema, exp.Schema):
             schema = schema.this
 
+        if materialized_properties:
+            partitioned_by = materialized_properties.pop("partitioned_by", None)
```

Comment:
Why do we not do this just for BQ? Do other engines support this in materialized views?  I bet postgres doesn't. So far our strategy has been ignoring these for engines that don't support them.

Eg. see the `_build_table_properties_exp` method. Btw, any chance we can reuse it to populate the properties?

Reply:
Snowflake supports cluster by: https://docs.snowflake.com/en/sql-reference/sql/create-materialized-view

Databricks supports partition by: https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-materialized-view.html

So I don't think it is correct to say that this is a BQ only feature.

Reply:
For the second part about `_build_table_properties_exp`, I actually initial did that: https://github.com/TobikoData/sqlmesh/pull/3117/commits/daf02da3fecc87b1337424a114ca2914028f8fc6

I then realized that the contract with that method is that it build table properties for a given engine therefore someone could include default table properties that wouldn't apply to a view. Clickhouse had some errors that made me realize that this wasn't the right pattern. Therefore I split specifically building cluster by/partition by since they were shared across tables and views.

Reply:
https://github.com/TobikoData/sqlmesh/pull/3117/commits/37ab52c49a691e28a287f86815b1a04275851f21